File: web3rpc/sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/debug/others/DebugStorageRangeAtTest.java
package opensdk.sdk.apis.debug.others;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

import java.io.IOException;
import opensdk.sdk.apis.constant.UrlConstants;
import org.web3j.protocol.klaytn.core.method.response.DebugStorageRangeAtResponse;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.web3j.protocol.klaytn.Web3j;
import org.web3j.protocol.http.HttpService;
@DisplayName("Debug RPC Test")
public class DebugStorageRangeAtTest {
  private Web3j w3 = Web3j.build(new HttpService(UrlConstants.SERVER_URL));
  @Disabled
  @Test
  @DisplayName("RPC debug_storageRangeAt")
  void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
    String blockHash = "0x90c81195698bc9f282bbdec386b0afb4dcc28e43aae834894281c3ecb3c88d21";
    int txIndex = 1;
    String contractAddress = "0x73a7d19d14f7dfac5b799e405e22133b2adc57a6";
    String keyStart = "0x12";
    int maxResult = 1;

    DebugStorageRangeAtResponse response = w3.debugStorageRangeAt(blockHash, txIndex, contractAddress, keyStart, maxResult).send();
    assertNotNull(response);
    assertNull(response.getError());
  }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/others/DebugStorageRangeAt.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk.DebugApi(new OpenSdk.ApiClient(RPC));

describe('debug_storageRangeAt API', () => {
    test.skip('should return debug_storageRangeAt', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            done();
        };

        const blockHash = "0x90c81195698bc9f282bbdec386b0afb4dcc28e43aae834894281c3ecb3c88d21";
        const txIndex = 1;
        const contractAddress = "0x73a7d19d14f7dfac5b799e405e22133b2adc57a6";
        const keyStart = "0x12";
        const maxResult = 1;

        sdk.storageRangeAt(blockHash, txIndex, contractAddress, keyStart, maxResult, {}, callbackOne);
    });
});

